### PR TITLE
Fix experiment logic in amp-carousel

### LINKED ISSUE #18558: amp-carousel: Janky on iOS 12
Seems mostly problematic on iframe-embedded AMP pages e.g. viewers and the ABE playground.

Repro cases:
- https://www.google.com/amp/s/www.mysanantonio.com/news/texas/amp/No-free-lunch-for-renewables-More-wind-power-13282311.php
- https://www.google.com/amp/s/ampbyexample.com/components/amp-carousel/
- https://ampbyexample.com/playground/#url=https%3A%2F%2Fampbyexample.com%2Fcomponents%2Famp-carousel%2Fsource%2F&preview=true

`b/116239446`

/to @aghassemi /cc @nainar

### PR DESCRIPTION
**Pull request by nainar**
Currently CSS Scroll Snap is disabled in all iOS platform:
The condition read:
```
this.shouldDisableCssSnap_ = !isExperimentOn(
        this.win, 'amp-carousel-scroll-snap') ||
        startsWith(
            Services.platformFor(this.win).getIosVersionString(),
            '10.3');
```
Since the experiment was turned off, the experiment check returned `false` which meant we were disabling CSS Scroll Snap on iOS - where it should have been on. 

Fixes #18558, #18592

### CODE REVIEW

**Comment by aghassemi** [comment]
@nainar Let's rename `amp-carousel-scroll-snap` to `amp-carousel-chrome-scroll-snap` to reduce the confusion around this flag.

**Comment by aghassemi** [comment]
@nainar can we also turn off the experiment in Canary? it is impacting 1% of prod sides and it doesn't seem to be working properly in Chrome. When I try m.aliexpress.com with Canary, carousel "double" jumps

**Comment by nainar** [comment]
@aghassemi  - renamed the experiment in this PR. 

Turning off the experiment in a separate PR.

File: tools/experiments/experiments.js [hunk 1]
```diff
@@ -329,7 +329,7 @@ const EXPERIMENTS = [
     cleanupIssue: 'https://github.com/ampproject/amphtml/issues/17243',
   },
   {
-    id: 'amp-carousel-scroll-snap',
+    id: 'amp-carousel-chrome-scroll-snap',
```
Comment (aghassemi) [inline tools/experiments/experiments.js]: s/`Enables scroll snap on carousel across all browsers/OSes`/ `Enables scroll snap on carousel on Chrome browsers`

Comment (aghassemi) [reply tools/experiments/experiments.js]: I guess a separate PR to remove the old name from Canary config.

**Pull request merged**